### ISSUE TITLE
Enforce newline after description but NOT when there is no description

### ISSUE THREAD

**Issue by yvele** [gajus/eslint-plugin-jsdoc#1033]
I want to update eslint-plugin-jsdoc from v41 to v43.
But `jsdoc/newline-after-description` has been removed.
I cannot reproduce the same behaviour with `jsdoc/tag-lines` as recommanded in the changelog.

## Expected behavior

```js
/**
 * Description.
 * 
 * @param {string} input
 */
function processSass (input) {
}
// "jsdoc/newline-after-description": "error"
```

And when there is NO DESCRIPTION I don't want an extra useless empty line (of course).
(Many time we don't have description as the function name is self explanatory)

```js
/**
 * @param {string} input
 */
function processSass (input) {
}
// "jsdoc/newline-after-description": "error"
```

## Actual behavior

How to avoid this useless extra empty line when there is no description?
(Many time we don't have description as the function name is self explanatory)

```js
/**
 *
 * @param {string} input
 */
function processSass (input) {
}
// "jsdoc/tag-lines": ["error", "never",{"startLines":1}]
```

## ESLint Config

```js
jsdoc/tag-lines
```

## Environment

- Node version: v18
- ESLint version v8.38.0
- `eslint-plugin-jsdoc` version: 43.0.6

**Comment by brettz9** [comment]
The rule as of 43.0.6 does not add whitespace when there is no description.

The following was not protected against by `newline-after-description` either:

```js
/**
 *
 * @param {string} input
 */
function processSass (input) {
}
```

If you want to prevent such scenarios, you can use the relatively new rule, `no-blank-block-descriptions`. This will complain about the above and fix it to:

```js
/**
 * @param {string} input
 */
function processSass (input) {
}
```

A bug in an earlier version of the `tag-lines` rule enforced a line when there is no description, so if you committed the changes while testing such a version and then upgraded to the latest version, the latest version of the rule does not fix that bug, but it doesn't introduce the bug either, so perhaps that is why you think the current version adds lines after a description?

**Comment by brettz9** [comment]
Closing as per the previous comment. However, please feel free to comment further as needed.

**Comment by yvele** [comment]
Thank you for your time @brettz9 it's working with `newline-after-description` 👍 

```js
/**
 * @param {string} input
 */
function processSass (input) {
}
// "jsdoc/no-blank-block-descriptions": "error",
// "jsdoc/tag-lines": ["error", "never",{"startLines":1}]
```

Not that I also needed `"jsdoc/multiline-blocks": "error"` to make sure there is no multi empty lines.

Thanks!

**Comment by brettz9** [comment]
> Thank you for your time @brettz9 it's working with `newline-after-description` 👍

Glad to hear it's working! I think you mean `no-blank-block-descriptions` as per the below--a nit I only mention in case you wouldn't mind editing the post for anyone finding the issue.

> Not that I also needed `"jsdoc/multiline-blocks": "error"` to make sure there is no **multi** empty lines.
> 
> Thanks!

Glad to hear, but unless I'm misunderstanding you, `no-blank-block-descriptions` should be the one responsible for that. Did you use `jsdoc/multiline-blocks` with any options? The `jsdoc/multiline-blocks` rule with the default settings should not impact whitespace. It's more for avoiding (or enforcing) text like:

```js
       /** desc
         *
         */
```

If you're looking at the test output for the rule, you might see some whitespace related options for the rule, but that is if you set the `noMultilineBlocks` option to `true`. However, I would only change the default for this option if you are sure you know what you are doing because this will entirely block any multiline blocks--not just ones with whitespace, e.g.

```js
       /** 
         * desc
         */
```

The option is used by those who want very dense JSDoc, e.g.,

```js
       /**  desc */
```
